mllib/src/main/scala/org/apache/spark/mllib/evaluation/RegressionMetrics.scala

```diff
@@ -31,7 +31,8 @@ import org.apache.spark.sql.DataFrame
  * @param predictionAndObservations an RDD of (prediction, observation) pairs.
  */
 @Experimental
-class RegressionMetrics(predictionAndObservations: RDD[(Double, Double)]) extends Logging {
+class RegressionMetrics(predictionAndObservations: RDD[(Double, Double)])
+  extends Logging with Serializable {
```

Comment:
Not too sure why but I was getting `Task Not Serializable` errors without this. I suspect this is because everything inside the `train()` method's closure gets serialized.

I followed up on how lazy vals interact with serialization and found [this SO post](http://stackoverflow.com/questions/27882307/how-does-serialization-of-lazy-fields-work) which says that the value is serialized iff it was computed before serialization.

In my updated implementation, one option could be to force the evaluation of `RegressionMetrics.summary` in the `LinearRegressionTestResults` constructor. However, despite being serializable I don't expect this class to be replicated anywhere except at the driver so maybe this eager evaluation is unnecessary. @jkbradley thoughts?